File: 1.two-sum/solution.js
const assert = require('node:assert/strict')

function twoSum(nums, target) {
  const map = {}
  for (const [index, num] of nums.entries()) {
    if (map[target - num] !== undefined) {
      return [map[target - num], index]
    }
    map[num] = index
  }
  return null
}


(function () {
  assert.deepEqual(twoSum([2, 7, 11, 15], 9), [0, 1])
  assert.deepEqual(twoSum([3, 2, 4], 6), [1, 2])
  assert.deepEqual(twoSum([1, 7, 7], 8), [0, 1])
  console.log("Accepted!")
})()

File: 1.two-sum/solution.go
package main

import "fmt"

func twoSum(nums []int, target int) []int {
	cache := map[int]int{}
	for index, num := range nums {
		if value, ok := cache[target - num]; ok {
			return []int{value, index}
		}
		cache[num] = index
	}
	return []int{}
}


func main() {
	res := twoSum([]int{3, 2, 1, 4}, 9)
	fmt.Println(res)
}